packages/core/src/utils/markers/index.ts

```diff
@@ -26,13 +27,28 @@ const makeMarker = ({
   fill = defaultFill,
 }: MarkerStyle) =>
   `${prefix}${encodeURIComponent(`
-<svg width="${imgSize[0]}" height="${imgSize[1]}" viewBox="0 0 30 43" xmlns="http://www.w3.org/2000/svg">
+<svg width="${imgSize[0]}" height="${
+    imgSize[1]
+  }" viewBox="0 0 30 43" xmlns="http://www.w3.org/2000/svg">
   <title>DB6C494E-88E8-49F1-89CE-97CBEC3A5240</title>
+  ${
+    typeof fill === 'string'
+      ? ''
+      : `<defs>
+          <pattern id="img" patternUnits="userSpaceOnUse" x="0" y="0" width="${
+            fill.size || 30
+          }" height="${fill.size || 30}">
+            <image href="${getPolygonFillHatch(fill).toDataURL()}" width="${
+          fill.size || 30
+        }" height="${fill.size || 30}"/>
+          </pattern>
+        </defs>`
```

Comment:
This code segment is added here twice.  
I suggest adding a function that receives `fill` as a parameter and returns this segment

Reply:
Good catch.

https://github.com/Dataport/polar/pull/81/commits/c7b6a7c31c4952c6d325854765758674f1de07b3